weblab_cg/tests/test_chaste_cg.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import os
+import re
+import weblab_cg as cg
+import pytest
+import cellmlmanip
+from weblab_cg.tests.chaste_test_utils import load_chaste_models, get_file_lines, write_file
+
+
+# Show more logging output
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.DEBUG)
+
+
+class TestChasteCG(object):
+    """ Tests weblab_cg against reference models generated with weblab_cg and tested in chaste.
+
+    TODO: unit conversion vvia cellmlmanip once implemented"""
+    _TIMESTAMP_REGEX = re.compile(r'(//! on .*\n)')
+
+    def model_types(self):
+        return ['Normal']
+
+    @pytest.fixture(scope="class")
+    def chaste_models(self):
+        """ Load all models"""
+        return load_chaste_models(model_types=self.model_types())
+
+    def test_generate_chatse_models(self, tmp_path, chaste_models):
+        """ Check generation of Normal models against reference"""
+        tmp_path = str(tmp_path)
+        for model in chaste_models:
```

Comment:
Rather than iterating here, you might want to investigate using pytest's test parameterisation to have a separate test case per model. But not in this PR!

Reply:
interesting idea I'll look at it at some point.